src/io.c

```diff
@@ -1,4478 +1,4641 @@
+            else
+                bytes_per_blockelement = ChimesGlobalVars.totalNumberOfSpecies * sizeof(MyOutputFloat);
+#endif
+            break;
+
+        case IO_CHIMES_REDUCED:
+            if(mode)
+                bytes_per_blockelement = 4 * sizeof(MyInputFloat);
+            else
+                bytes_per_blockelement = 4 * sizeof(MyOutputFloat);
+            break;
+
+        case IO_CHIMES_FLUX_G0:
+        case IO_CHIMES_FLUX_ION:
+#if defined(CHIMES) && defined(CHIMES_STELLAR_FLUXES)
+            if(mode)
+                bytes_per_blockelement = CHIMES_LOCAL_UV_NBINS * sizeof(MyInputFloat);
+            else
+                bytes_per_blockelement = CHIMES_LOCAL_UV_NBINS * sizeof(MyOutputFloat);
+#endif
+            break;
+
+        case IO_CBE_MOMENTS:
+
+        case IO_EOS_STRESS_TENSOR:
+        case IO_TIDALTENSORPS:
+        case IO_SHEET_ORIENTATION:
+            if(mode)
+                bytes_per_blockelement = 9 * sizeof(MyInputFloat);
+            else
+                bytes_per_blockelement = 9 * sizeof(MyOutputFloat);
+            break;
+
+        case IO_GDE_DISTORTIONTENSOR:
+            if(mode)
+                bytes_per_blockelement = 36 * sizeof(MyInputFloat);
+            else
+                bytes_per_blockelement = 36 * sizeof(MyOutputFloat);
+            break;
+
+        case IO_LAST_CAUSTIC:
+            if(mode)
+                bytes_per_blockelement = 20 * sizeof(MyInputFloat);
+            else
+                bytes_per_blockelement = 20 * sizeof(MyOutputFloat);
+            break;
+
+        case IO_LASTENTRY:
+            endrun(214);
+            break;
+    }
+
+    return bytes_per_blockelement;
+}
+
+int get_datatype_in_block(enum iofields blocknr)
+{
+    int typekey;
+    switch (blocknr)
+    {
+#if (defined(OUTPUT_POSITIONS_IN_DOUBLE) && !defined(OUTPUT_IN_DOUBLEPRECISION)) || (defined(INPUT_POSITIONS_IN_DOUBLE) && !defined(INPUT_IN_DOUBLEPRECISION))
+        case IO_POS:
+            typekey = 3; /* pos outputs in HDF5 are double automatically, to prevent overlaps */
+            break;
+#endif
+
+        case IO_ID:
+        case IO_CHILD_ID:
+#ifdef LONGIDS
+            typekey = 2;		/* native long long */
+#else
+            typekey = 0;		/* native int */
+#endif
+            break;
+
+        case IO_GENERATION_ID:
+#if defined(BH_WIND_SPAWN) && defined(LONGIDS) //we use this to store progenitor info so this needs to be able to handle any valid ID, rather than the usual 0-32
+            typekey = 2;		/* native long long */
+#else
+            typekey = 0;		/* native int */
+#endif
+            break;
+
+        case IO_BHPROGS:
+        case IO_GRAINTYPE:
+        case IO_EOSCOMP:
+        case IO_STAGE_PROTOSTAR:
+            typekey = 0;		/* native int */
+            break;
+
+        case IO_SLUG_STATE_RNG:
+            typekey = 4;		/* 128 int * Define later what I should do with this case. 2 Places */

```

Comment:
The datatype for this may be tricky. You may have to write this to the file as two 64-bit integers instead of one 128-bit integer, but I'm not sure what is the best strategy right now. I'll look into this.

Reply:
Yes, you should extract this into 2 separate 64-bit integers. I've added an example of how to do that in a previous comment.

Reply:
For reference, the correct HDF5 datatype is `H5T_NATIVE_ULLONG`.

Reply:
`typekey = 2;` here